doc(XCP-NG): replace discontinued FreeNAS / TrueNAS CORE by fork zVault (#373)
xcp-ng/xcp-ng-org: docs/vms/vms.md

### FreeBSD Guest Tools

> FreeBSD is a 30-year-old operating system used widely to run all sorts of systems and has served as the basis for a number of operating systems, including MacOS, pfSense / OPNsense and FreeNAS->TrueNAS->zVault (see appliances further bellow). The Xen kernel modules are built and distributed in the GENERIC kernel, so if you haven't customised or recompiled your kernel, the drivers will be present.

Reviewer: ```suggestion
FreeBSD is a 30-year-old operating system used widely to run all sorts of systems and has served as the basis for a number of operating systems, including MacOS, pfSense / OPNsense and FreeNAS->TrueNAS->zVault (see appliances further below). The Xen kernel modules are built and distributed in the GENERIC kernel, so if you haven't customised or recompiled your kernel, the drivers will be present.
```

Author: Thanks for catching this :-)

For OpenBSD search [the forum](https://xcp-ng.org/forum). See for example [this thread](https://xcp-ng.org/forum/topic/2582/guest-tools-for-openbsd).
:::

### FreeBSD based appliances

#### pfSense / OPNsense

See this [section](/guides/pfsense/#2-install-guest-utilities) in dedicated guide.

> #### (formerly) FreeNAS (then) TrueNAS CORE (now) TrueNAS Legacy and (possible continuation) zVault

Reviewer: Simpler titles make for simpler anchor links
```suggestion
#### zVault
A fork of TrueNAS CORE (formerly FreeNAS), created when development stopped on TrueNAS CORE.
```

For OpenBSD search [the forum](https://xcp-ng.org/forum). See for example [this thread](https://xcp-ng.org/forum/topic/2582/guest-tools-for-openbsd).
:::

### FreeBSD based appliances

> #### pfSense / OPNsense Guest Tools

Author: Thanks :-)

See this [section](../guides/pfsense/#2-install-guest-utilities) in dedicated guide.

#### FreeNAS/TrueNAS Guest Tools

> Note: iXsystems' TrueNAS Legacy 13.3 is [final release](https://www.truenas.com/docs/core/13.3/gettingstarted/corereleasenotes/#133-u12-changelog) of their FreeBSD based line, being in (some) [maintenace](https://www.truenas.com/blog/truenas-core-13-3-plans/) only mode, but limited by EoL of FreeBSD 13.x on [April 30, 2026](https://www.freebsd.org/security/#sup) (please don't confuse with TrueNAS CORE Enterprise, which have [different EoL](https://www.truenas.com/docs/core/13.0/gettingstarted/corereleasenotes/), but as tied to hardware, thus irelevant to virtualization).

Reviewer 2: ```suggestion
Note: iXsystems' TrueNAS Legacy 13.3 is [final release](https://www.truenas.com/docs/core/13.3/gettingstarted/corereleasenotes/#133-u12-changelog) of their FreeBSD-based line, being in (some) [maintenace](https://www.truenas.com/blog/truenas-core-13-3-plans/) only mode, but limited by EoL of FreeBSD 13.x on [April 30, 2026](https://www.freebsd.org/security/#sup) (please don't confuse with TrueNAS CORE Enterprise, which have [different EoL](https://www.truenas.com/docs/core/13.0/gettingstarted/corereleasenotes/), but as tied to hardware, thus irelevant to virtualization).
```

Reviewer 2: Yeah, this sounds beter. How about _"TrueNAS Legacy 13.3 is the [final release](https://www.truenas.com/docs/core/13.3/gettingstarted/corereleasenotes/#133-u12-changelog) of their FreeBSD-based line"_

and:

_"Please don't confuse with TrueNAS CORE Enterprise, which has a [different EoL](https://www.truenas.com/docs/core/13.0/gettingstarted/corereleasenotes/), but since it is tied to hardware, it is irrelevant to virtualization."_?

> Note: iXsystems' TrueNAS Legacy 13.3 is the [final release](https://www.truenas.com/docs/core/13.3/gettingstarted/corereleasenotes/#133-u12-changelog) of their FreeBSD based line. It is in (some) [maintenace](https://www.truenas.com/blog/truenas-core-13-3-plans/) only mode, but limited by EoL of FreeBSD 13.x on [April 30, 2026](https://www.freebsd.org/security/#sup). Please don't confuse with TrueNAS CORE Enterprise, which has a [different EoL](https://www.truenas.com/docs/core/13.0/gettingstarted/corereleasenotes/), but since it is tied to hardware, it is irrelevant to virtualization.

Reviewer 2: ```suggestion
Note: iXsystems' TrueNAS Legacy 13.3 is the [final release](https://www.truenas.com/docs/core/13.3/gettingstarted/corereleasenotes/#133-u12-changelog) of their FreeBSD based line. It is in (some) [maintenance](https://www.truenas.com/blog/truenas-core-13-3-plans/) only mode, but limited by EoL of FreeBSD 13.x on [April 30, 2026](https://www.freebsd.org/security/#sup). Please don't confuse with TrueNAS CORE Enterprise, which has a [different EoL](https://www.truenas.com/docs/core/13.0/gettingstarted/corereleasenotes/), but since it is tied to hardware, it is irrelevant to virtualization.
```

> Since versions 12.0-U1 of this OS, the package is preinstaled by default (see this [comment](https://github.com/xcp-ng/xcp/issues/446#issuecomment-720632737)). In more recent releases is even starting necessary daemon at boot preconfigured - so everything is working out of box. In case of older (but since 12.0-U1) releases, just create rc.conf item `xenguest_enable` set to YES (see [documentation](https://www.truenas.com/docs/core/13.3/coretutorials/systemconfiguration/configuringtunables/)) - then either reboot or, if you do not plan to reboot the VM now, you can start the daemon manually running the command `/usr/local/sbin/xe-daemon -p /var/run/xe-daemon.pid &`). After you'll see a FreeBSD icon in your VM list on Xen Orchestra. Since then you can (among other things) restart/shutdown the VM properly from the XO (Lite) Web UI.

Author: It means, that from 12.0-U1 onward was guest tools package installed by default, but service daemon was not started by default - user had to add it into rc.conf configuration file. But in more recent versions this config is already there by default. Developers perhaps realized, that it is no problem if this service is running even in cases, when TrueNAS is not installed in XEN VM (eg. on bare metal).

> Since versions 12.0-U1 of this OS, the package is preinstaled by default (see this [comment](https://github.com/xcp-ng/xcp/issues/446#issuecomment-720632737)). In more recent releases is even starting necessary daemon at boot preconfigured - so everything is working out of box. In case of older (but since 12.0-U1) releases, just create rc.conf item `xenguest_enable` set to YES (see [documentation](https://www.truenas.com/docs/core/13.3/coretutorials/systemconfiguration/configuringtunables/)) - then either reboot or, if you do not plan to reboot the VM now, you can start the daemon manually running the command `/usr/local/sbin/xe-daemon -p /var/run/xe-daemon.pid &`). After you'll see a FreeBSD icon in your VM list on Xen Orchestra. Since then you can (among other things) restart/shutdown the VM properly from the XO (Lite) Web UI.

Author: Sounds good - incorporated into latest edits. Thanks :-)

> Since versions 12.0-U1 of this OS, the package is preinstaled by default (see this [comment](https://github.com/xcp-ng/xcp/issues/446#issuecomment-720632737)). In more recent releases, the necessary daemon starts automatically when the system boots, so everything works out of the box. In case of older (but since 12.0-U1) releases, just create rc.conf item `xenguest_enable` set to YES (see [documentation](https://www.truenas.com/docs/core/13.3/coretutorials/systemconfiguration/configuringtunables/)) - then either reboot or, if you do not plan to reboot the VM now, you can start the daemon manually running the command `/usr/local/sbin/xe-daemon -p /var/run/xe-daemon.pid &`). After you'll see a FreeBSD icon in your VM list on Xen Orchestra. Since then you can (among other things) restart/shutdown the VM properly from the XO (Lite) Web UI.

Author: Thanks for catching also this :-)

> Since versions 12.0-U1 of this OS, the package is preinstalled by default (see this [comment](https://github.com/xcp-ng/xcp/issues/446#issuecomment-720632737)). In more recent releases, the necessary daemon starts automatically when the system boots, so everything works out of the box. In case of older (but since 12.0-U1) releases, just create rc.conf item `xenguest_enable` set to YES (see [documentation](https://www.truenas.com/docs/core/13.3/coretutorials/systemconfiguration/configuringtunables/)) - then either reboot or, if you do not plan to reboot the VM now, you can start the daemon manually running the command `/usr/local/sbin/xe-daemon -p /var/run/xe-daemon.pid &`). After you'll see a FreeBSD icon in your VM list on Xen Orchestra. Since then you can (among other things) restart/shutdown the VM properly from the XO (Lite) Web UI.

Author: Thanks :-)

> More insights and options (for older versions) are available in [this issue](https://github.com/xcp-ng/xcp/issues/172#issuecomment-548181589) or [this issue](https://github.com/xcp-ng/xcp/issues/446).

Reviewer 2: Yes, my bad. Let's use your suggestion!

> More insights and options (for older versions) are available in [this issue](https://github.com/xcp-ng/xcp/issues/172#issuecomment-548181589) or [this issue](https://github.com/xcp-ng/xcp/issues/446).

Author: That's okay - thanks :-)

#### zVault
A [fork](https://zvault.io/) of TrueNAS CORE (formerly FreeNAS), created when development stopped on TrueNAS CORE.

> As zVault started as fork of TrueNAS Legacy 13.x, all is working out of box, as stated in FreeNAS/TrueNAS section above (and this probably won't change even in upcoming major releases).

Author: Thanks :-)
